File: translations/codenet/mistral_commenetd/C2P/gpt-4o/Code/s484809551.py
def strrstr(string, pattern):
    """Find the last occurrence of a substring in a string."""
    last_index = -1
    offset = 0

    while True:
        index = string.find(pattern, offset)
        if index == -1:
            break
        last_index = index
        offset = index + 1

    return last_index


def main():
    # Define pattern base string
    pattern_base = "keyence"
    S = input().strip()

    # Calculate length of input string
    len_S = len(S)
    found = False

    # Loop through each character in pattern base string
    for offset in range(len(pattern_base) + 1):
        # Copy substring up to current character to top variable
        top = pattern_base[:offset]

        # Copy substring from current character to end of pattern base string to bottom variable
        bottom = pattern_base[offset:]

        # Search for top substring in input string
        top_search = S.find(top)

        # If top substring is found
        if top_search != -1:
            # Search for bottom substring after top substring in input string
            bottom_search = strrstr(S[top_search + offset:], bottom)

            # If bottom substring is found
            if bottom_search != -1:
                # Calculate number of separators between top and bottom substrings
                sep_count = 0

                # If top_search is not the first occurrence of top substring in input string
                if top_search != 0:
                    sep_count += 1

                # If top_search is not the same as bottom_search
                if top_search + offset != bottom_search + top_search + offset:
                    sep_count += 1

                # If bottom_search is not the last character in input string
                if bottom_search + len(bottom) < len_S:
                    sep_count += 1

                # If number of separators is less than or equal to 1
                if sep_count <= 1:
                    found = True
                    break

    # Print output based on whether pattern is found or not
    print("YES" if found else "NO")


if __name__ == "__main__":
    main()

# <END-OF-CODE>

File: translations/codenet/mistral_commenetd/C2Go/gpt-4o/Code/s484809551.go
package main

import (
	"fmt"
	"strings"
)

// Define constants
const (
	N_MAX = 100
	P_MAX = 100
)

// Function to find the last occurrence of a substring in a string
func strrstr(s, pattern string) string {
	lastIndex := -1
	for i := 0; i <= len(s)-len(pattern); i++ {
		if s[i:i+len(pattern)] == pattern {
			lastIndex = i
		}
	}
	if lastIndex == -1 {
		return ""
	}
	return s[lastIndex:]
}

// Function to calculate the number of combinations of n items taken r at a time
func comb(n, r int) int {
	result := 1
	for i := 0; i < r; i++ {
		result *= n - i
		result /= i + 1
	}
	return result
}

// Function to find the greatest common divisor of two numbers
func gcd(m, n uint64) uint64 {
	var temp uint64
	for m%n != 0 {
		temp = n
		n = m % n
		m = temp
	}
	return n
}

func main() {
	// Define pattern base string
	patternBase := "keyence"
	var S string

	// Read input string from standard input
	fmt.Scan(&S)

	// Calculate length of input string
	lenS := len(S)
	found := false

	// Loop through each character in pattern base string
	for offset := 0; offset < len(patternBase); offset++ {
		// Copy substring up to current character to top variable
		top := patternBase[:offset]

		// Copy substring from current character to end of pattern base string to bottom variable
		bottom := patternBase[offset:]

		// Search for top substring in input string
		topSearch := strings.Index(S, top)

		// If top substring is found
		if topSearch != -1 {
			// Search for bottom substring after top substring in input string
			bottomSearch := strrstr(S[topSearch+offset:], bottom)

			// If bottom substring is found
			if bottomSearch != "" {
				// Calculate number of separators between top and bottom substrings
				sepCount := 0

				// If topSearch is not the first occurrence of top substring in input string
				if topSearch != 0 {
					sepCount++
				}

				// If topSearch is not the same as bottomSearch
				if S[topSearch+offset:] != bottomSearch {
					sepCount++
				}

				// If bottomSearch is not the last character in input string
				if bottomSearch[len(bottom):] != "" {
					sepCount++
				}

				// If number of separators is less than or equal to 1
				if sepCount <= 1 {
					found = true
					break
				}
			}
		}
	}

	// Print output based on whether pattern is found or not
	if found {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

// <END-OF-CODE>
